### PR TITLE
[JENKINS-41729] update to new localizer plugin version

### PR DESCRIPTION
**Pull request by jtnord**
Downstream of https://github.com/jenkinsci/pom/pull/150

which is in turn downstream of
>  updates the localizer plugin to 1.28 (adds threadsagety support for parallel builds, and messages from XML) kohsuke/localizer@localizer-parent-1.26...localizer-parent-1.28

*NOTE*: this also pick up a change in localizer to allow definition of messages in XML files as well as properties, bit *I* have not tested that and there is discussion in the Jira if we actually would want that.  I leave that to others to decide if we want to mention it in the release notes.  The main change required here was that localizer was not thread safe and thus threaded maven builds (`-T`) could (in theory but not observed) produce corrupted output at worst and a best produced an annoying warning at build time

[JENKINS-41729](https://issues.jenkins.io/browse/JENKINS-41729)

### Proposed changelog entries

* Developer: update localizer plugin to allow threaded builds and localization from XML as well as properties

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@ THE SOFTWARE.
   <parent>
     <groupId>org.jenkins-ci</groupId>
     <artifactId>jenkins</artifactId>
-    <version>1.58</version>
+    <version>1.61-20201201.120825-2</version>
```
Comment (jtnord) [inline pom.xml:31]: obviously waiting a merge and release of  https://github.com/jenkinsci/pom/pull/150

File: cli/pom.xml [hunk 2]
```diff
@@ -68,7 +68,6 @@
     <dependency>
       <groupId>org.jvnet.localizer</groupId>
       <artifactId>localizer</artifactId>
-      <version>1.26</version>
```
Comment (jtnord) [inline cli/pom.xml:71]: pick up the version from the bom

File: pom.xml (hunk 1 above)
Comment (jtnord) [reply pom.xml:31]: now released 
```suggestion
    <version>1.61</version>
```

**Review by timja — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by StefanSpieker — APPROVED** [review]

**Pull request merged**